docs/extensions/sandbox/sandbox-sdk.md

```diff
@@ -0,0 +1,212 @@
+
+import type { SandboxEndpointRouter } from 'directus:api';
+import { request } from "directus:api";
+
+export default (router: SandboxEndpointRouter) => {
+	router.get("/hello", async () => {
+		const response = await request("https://directus.io", {
+			method: 'POST'
+		});
+
+		return {
+			status: 200,
+			body: response.data,
+		};
+	});
+};
+```
+
+## Reference
+
+The following SDK functions are currently available:
+
+- [Log](#log)
+- [Sleep](#sleep)
+- [Request](#request)
+
+### Log
+
+The `log` function will print a message in the API's logger output. It can be used as a replacement for `console.log`.
+
+::: tabs
+
+== Signature
+
+```ts
+type SandboxLogFn = (message: string) => void;
+```
+
+== Example
+
+```js
+import { log } from 'directus:api';
+
+log('Hello World!');
+```
+
+:::
+
+#### Required Scopes
+
+In order to use `log`, your extension needs to have the `log` property present in the requested scopes:
+
+```json{9}
+"directus:extension": {
+	"type": "endpoint",
+	"path": "dist/index.js",
+	"source": "src/index.js",
+	"host": "^10.7.0",
+	"sandbox": {
+		"enabled": true,
+		"requestedScopes": {
+			"log": {}
+		}
+	}
+}
+```
+
+---
+
+### Sleep
+
+The `sleep` function will wait for a given amount of time. It can be used as a replacement for `setTimeout`.
+
+::: tabs
+
+== Signature
+
+```ts
+type SandboxSleepFn = (milliseconds: number) => Promise<void>;
+```
+
+== Example
+
+```js
+import { sleep } from 'directus:api';
+
+await sleep(1000);
+```
+
+:::
+
+#### Required Scopes
+
+In order to use `sleep`, your extension needs to have the `sleep` property present in the requested scopes:
+
+```json{9}
+"directus:extension": {
+	"type": "endpoint",
+	"path": "dist/index.js",
+	"source": "src/index.js",
+	"host": "^10.7.0",
+	"sandbox": {
+		"enabled": true,
+		"requestedScopes": {
+			"sleep": {}
+		}
+	}
+}
+```
+
+---
+
+### Request
+
+The `request` function will make a network request to the given URL.
+
+::: tabs
+
+== Signature
+
+```ts
+interface SandboxRequestResponse {
+	status: number;
+	statusText: string;
+	headers: Record<string, string>;
+	data: string | Record<string, unknown>;
+}
+
+type SandboxRequestFn = (
+	url: string,
+	options: {
+		method?: 'GET' | 'POST' | 'PUT' | 'PATCH' | 'DELETE';
+		body?: Record<string, any> | string;
+		headers?: Record<string, string>;
+	}
+) => Promise<SandboxRequestResponse>;
+```
+
+== Example
+
+```js
+import { request } from 'directus:api';
+
+const response = await request('https://directus.io');
+```
+
+:::
+
+#### Required Scopes
+
+In order to use `request`, your extension needs to have the `request` property present in the requested scopes,
+specifying the methods and URLs you want to use:
+
+```json{9-12}
+"directus:extension": {
+	"type": "endpoint",
+	"path": "dist/index.js",
+	"source": "src/index.js",
+	"host": "^10.7.0",
+	"sandbox": {
+		"enabled": true,
+		"requestedScopes": {
+			"request": {
+				"methods": ["GET"],
+				"urls": ["https://directus.io"]
```

Comment:
Are wildcards still allowed? If so, should be mentioned along with an example.

Reply:
https://github.com/directus/directus/pull/20116 they are now 👍🏻

Reply:
Awesome ❤️